tests/integration_test.rs

```diff
@@ -0,0 +1,65 @@
+extern crate etherparse;
+extern crate serial_test;
+extern crate tun_tap;
+
+use etherparse::{IpHeader, PacketBuilder, PacketHeaders, TransportHeader};
+use serial_test::serial;
+use std::net::{IpAddr, Ipv4Addr, UdpSocket};
+use tun_tap::{Iface, Mode};
+
+#[test]
+#[serial]
+fn it_sents_packets() {
+    let iface =
+        Iface::without_packet_info("tun0", Mode::Tun).expect("failed to create a TUN device");
+    let data = [1; 10];
+    let socket = UdpSocket::bind("10.10.10.1:2424").expect("failed to bind to address");
+    socket
+        .send_to(&data, "10.10.10.2:4242")
+        .expect("failed to send data");
+    let mut buf = [0; 50];
+    let num = iface.recv(&mut buf).expect("failed to receive data");
+    assert_eq!(num, 38);
+    let packet = &buf[..num];
+    if let PacketHeaders {
+        ip: Some(IpHeader::Version4(_ip_header)),
+        transport: Some(TransportHeader::Udp(_udp_header)),
+        payload,
+        ..
+    } = PacketHeaders::from_ip_slice(&packet).expect("failed to parse packet")
+    {
+        assert_eq!(_ip_header.source, [10, 10, 10, 1]);
+        assert_eq!(_ip_header.destination, [10, 10, 10, 2]);
+        assert_eq!(_udp_header.source_port, 2424);
+        assert_eq!(_udp_header.destination_port, 4242);
```

Comment:
Looking at this again, why are the variable names with `_`? Using them in assert counts as use, so you shouldn't get an unused variable lint, and *especially* in tests, checking them is the intended use.

Reply:
I got compiler warnings for having variable names like `ip_header`. `_` prefix was the compiler proposed. After upgraded to 1.40 the warnings were gone.